content/articles/diogogpinto-filament-themes-full-guide.md

```diff
@@ -0,0 +1,333 @@
+
+## Customizing the theme
+
+Laravel, and Filament being Laravel based, offer multiple solutions to the same problem. So, we will address two ways of building your own Filament look and feel, the recommended way and the not-so-recommended way. Let's start with the latter.
+
+### Before starting the customization process
+
+You should start the `npm run dev` process in the root of your project. This process watches for changes in the files registered in the `vite.config.js` (and `tailwind.config.js` by extension) and automatically builds a temporary CSS file to render the changes on file save.
+
+> [!note]
+> You still need to run `npm run build` when you're done customizing your theme
+
+### The recommended way to customize your theme
+
+The Filament panels core offers you a list of classes. Every component of the panel layout, including forms, infolists and tables, were wrapped in custom classes to make it possible, for any developer, to customize the panel look and feel. Let's look at the following example:
+
+![Breadcrumb Example](/images/content/articles/diogogpinto-filament-themes-full-guide/breadcrumb-example.webp)
+
+Looking at the breadcrumb component with Chrome's builtin Developer Tools > Inspect Element, you can see that the `<nav>`na tag contains the following classes:
+
+```html
+<nav class="fi-breadcrumbs mb-2 hidden sm:block">
+```
+
+The `.fi-breadcrumbs` class is inserted to allow developers to further optimize the breadcrumbs' look and feel. By default, the class has no CSS properties defined in itself, it's only there for the developer to apply custom styles.
+
+#### Customizing the CSS
+
+Picking up in the above example of `.fi-breadcrumbs`, we can now apply custom CSS classes on our `theme.css`. You can either use Tailwind utility classes or just write vanilla CSS. Let's say you want to override the default margin `mb-2`. Inputting the below CSS will yield the exact same results:
+
+```css
+.fi-breadcrumbs {
+    @apply mb-4;
+}
+
+.fi-breadcrumbs {
+    margin-bottom: 1rem;
+}
+```
+
+As the `.fi-breadcrumbs` class is inserted before the default Tailwind utility classes, applying custom CSS will always override the default style.
+
+#### Diving deeper
+
+If we dive deeper into the breadcrumbs, we will find all the following classes, in this hierarchy:
+
+```
+- .fi-breadcrumbs
+-- .fi-breadcrumbs-list
+--- .fi-breadcrumbs-item
+---- .fi-breadcrumbs-item-separator
+---- .fi-breadcrumbs-item-label
+```
+
+Let's say we want to really change the breadcrumbs appearance. We can style each of this classes. Check the following example:
+
+```css
+.fi-breadcrumbs-list {
+    @apply gap-x-0;
+}
+
+.fi-breadcrumbs-item {
+    @apply gap-x-0;
+}
+
+.fi-breadcrumbs-item-label {
+    @apply font-thin dark:text-primary-500 text-primary-500;
+}
+
+span.fi-breadcrumbs-item-label {
+    @apply border-b-2 border-b-primary-500 font-semibold -mb-[2px];
+}
```

Comment:
It is confusing as to why you have the same class twice, with one being more specific due to the `span` selector. Could you combine these two together?

Reply:
@alexandersix I believe in this case breadcrumbs active item is wrapped in a `<span>` tag, the others on a `<a>` tag.

Maybe something like the following would be better?

```css
.fi-breadcrumbs-list > li:last-child .fi-breadcrumbs-item-label
```

Will have to try it out.